fixing reload loop, sbt refactor

Angel-O opened the pull request:
### Summary

- fixing reload loop
- sbt refactor


### Notes

- reload loop was caused by manual aggregation of projects, using sbt aggregation fixed the issue and drastically improved performance (`localPublish` script is up to 3 times faster 🚀 )
- using sbt aggregation was causing the `benchamark` project to fail during the `test` task. (I've made changes to ignore the task for that project, if it's not good we can find another solution)
- `commands.sbt` contains many aliases (@davesmith00000 I haven't checked all of them, let me know if you experience weird behaviour)

Angel-O commented on indigo/commands.sbt:
```diff
@@ -208,26 +207,20 @@ addCommandAlias(
 addCommandAlias(
   "crossLocalPublishNoClean",
-  List(
-    "crossBuildIndigo",
```
Removed it, because looking at the logs I think the `publish` task depends on the `compile` task

Angel-O commented on indigo/project/Misc.scala:
```diff
@@ -0,0 +1,66 @@
+    UsefulTask("", "code", "Launch VSCode")
+  )
+
+  def presentationSettings(version: SettingKey[String]): Seq[Def.Setting[String]] = {
```
called this `logoSettings` in Tyrian, I think this is a better name, feel free to rename it on `Tyrian` !

davesmith00000 commented:
Thanks @Angel-O 😄 

I was thinking about a lot of this recently, particularly in regard to the manual aggregation stuff. Mostly I'm using it out of habit now, but it came from the days when Indigo was cross published to JS + JVM, and I wanted a subsets of the build.

Long overdue a rethink...

I'll pull this down and try it out shortly. 👍

bilki commented on indigo/build.sbt:
```diff
@@ -1,40 +1,39 @@
   scalafixOnCompile  := true,
   semanticdbEnabled  := true,
   semanticdbVersion  := scalafixSemanticdb.revision,
   autoAPIMappings    := true
```
SBT welcome shows an ugly `Example` logo for subprojects otherwise.

```suggestion
  autoAPIMappings    := true,
  logo               := name.value
```

bilki reviewed (commented):
Nice refactor, it looks really tidy now. I keep hitting a warning though:

```
[warn] there's a key that's not used by any other settings/tasks:
[warn]
[warn] * indigoShaders / Test / scalaJSLinkerConfig
[warn]   +- ~/indigo/indigo/build.sbt:151
```

Also, it would be nice to run the source code formatter as well for these files (`scalafmtSbt` or just format document in vscode with metals). @davesmith00000 what do you think about adding the `scalafmt` sbt plugin?

Angel-O commented on indigo/commands.sbt:
```diff
@@ -102,10 +104,7 @@ addCommandAlias(
 addCommandAlias(
   "localPublishIndigo",
-  applyCommand(
-    coreProjects.filterNot(name => name == "sandbox" || name == "perf"),
```
I think this is no longer needed as the `neverPublish` settings in the `build.sbt` file should take care of not publishing artefacts for these projects.

Angel-O replied:
actually the current implementation is still trying to publish them for some reason 🤔  
at least according to the logs. Looking into it

davesmith00000 replied:
I just noticed that (might be an older version of this PR) a bunch of modules that should not be published (because they aren't artefacts) were published, like the benchmarks project.

Final list in you local ivy should be indigo, extras, json, and the plugin artefacts (indigo-plugin 2.12 + 2.13, mill-indigo, sbt-indigo).

Angel-O replied:
oh yeah, I think it's because I had `skip /  publish := true`, instead of `publish / skip : true`. I've double-checked now and I can see the correct list is being published. (Indigo is being published twice, adding the `neverPublish` settings to the aggregate should fix this)

Angel-O commented:
> Nice refactor, it looks really tidy now. I keep hitting a warning though:
> 
> ```
> [warn] there's a key that's not used by any other settings/tasks:
> [warn]
> [warn] * indigoShaders / Test / scalaJSLinkerConfig
> [warn]   +- ~/indigo/indigo/build.sbt:151
> ```
> 
> Also, it would be nice to run the source code formatter as well for these files (`scalafmtSbt` or just format document in vscode with metals). @davesmith00000 what do you think about adding the `scalafmt` sbt plugin?

I've removed the `Test / scalaJSLinkerConfig ~= { _.withModuleKind(ModuleKind.CommonJSModule) }` setting in the indigo-shaders project to fix the warning. Apparently it wasn't used so hopefully it's not a problem. Also I've formatted the following files with `vscode`

- build.sbt
- Misc.scala
- Dependencies.scala

I've also addressed the other comments. Happy to add a `scalafmt` file, what template should I use ?

bilki commented:
There's already a scalafmt file here https://github.com/PurpleKingdomGames/indigo/blob/master/indigo/.scalafmt.conf if you meant that 🤔

davesmith00000 commented:
@Angel-O ...I think it's time. If you would like to burn the `indigo-shaders` project with 🔥, be my guest. It's kind of a failed experiment that's been hanging around causing problems and adding no benefit for quite a long time now. 😂

davesmith00000 commented:
This is looking excellent. I did try it out earlier and it seemed to be doing the right things.

Thanks for joining in the review @bilki.

@Angel-O when you're happy (or have gone as far as you want to go), let me know and I'll give it another try. Would be great to merge this in before I have to do another release! 😅

bilki reviewed (dismissed):
I'd say this is ready to :shipit:

davesmith00000 commented:
@Angel-O You have removed the build references to indigo-shaders but not the folder with the code in it. 😋

Angel-O commented:
gone 🔥
